src/utils/variable_manipulations.jl

```diff
@@ -177,214 +180,377 @@ foreach_gs_tracer(f::F, Y_or_similar_values...) where {F} =
+  values of `a`.
+- For small `a_half`, the weight rapidly approaches 1 for values of `a` that are
+  a few times larger than `a_half`.
+
+**Construction Method:**
+The function is piecewise. For `a` between 0 and 1, it is a custom sigmoid curve
+constructed in two main steps to satisfy the key properties:
+1.  **Bounded Sigmoid Creation**: A base sigmoid is created that maps the interval
+    `(0, 1)` to `(0, 1)` with zero derivatives at the endpoints. This is achieved
+    by composing a standard `tanh` function with the inverse of a slower-growing
+    `tanh` function.
+2.  **Midpoint Control**: To ensure the function passes through the control point
+    `(a_half, 0.5)`, the input `a` is first transformed by a specially designed
+    power function (`1 - (1 - a)^k`) before being passed to the bounded sigmoid.
+    This transformation maps `a_half` to `0.5` while preserving differentiability 
+    at the boundaries.
+
+Arguments:
+- `a`: The input SGS area fraction (often approximated as `ρa / ρ`).
+- `a_half`: The value of `a` at which the weight function should be 0.5, controlling
+          the transition point of the sigmoid curve.
+
+Returns:
+- The computed weight, a value between 0 and 1.
+"""
+function sgs_weight_function(a, a_half)
+    if a < 0
         zero(a)
     elseif a > min(1, 42 * a_half) # autodiff generates NaNs when a is large
         one(a)
     else
         (1 + tanh(2 * atanh(1 - 2 * (1 - a)^(-1 / log2(1 - a_half))))) / 2
     end
+end
 
 """
-    divide_by_ρa(ρaχ, ρa, ρχ, ρ, turbconv_model)
+    draft_sum(f, sgsʲs)
 
-Computes `ρaχ / ρa`, regularizing the result to avoid issues when `a` is small.
-This is done by performing a linear interpolation from `ρaχ / ρa` to `ρχ / ρ`,
-using `sgs_weight_function(ρa / ρ, turbconv_model.a_half)` as the weight of
-`ρaχ / ρa` in the interpolation. Note that `ρa / ρ` is the "anelastic
-approximation" of `a`; we cannot directly use `a` to compute the weight because
-this function needs to be called before `a` has been computed. Also, note that
-using this function instead of directly computing `ρaχ / ρa` breaks the
-assumption of domain decomposition when the approximated `a` is small.
-"""
-function divide_by_ρa(ρaχ, ρa, ρχ, ρ, turbconv_model)
-    weight = sgs_weight_function(ρa / ρ, turbconv_model.a_half)
-    # If ρa = 0, we know that ρa / ρ = 0, which means that weight = 0. However,
-    # 0 * ρaχ / 0 = NaN, regardless of what ρaχ is, so the linear interpolation
-    # will always return NaN when ρa = 0. To avoid this problem, we need to add
-    # a special case for ρa = 0.
-    return ρa == 0 ? ρχ / ρ : weight * ρaχ / ρa + (1 - weight) * ρχ / ρ
-end
+Computes the sum of a function `f` applied to each draft subdomain
+state `sgsʲ` in the iterator `sgsʲs`.
 
+Arguments:
+- `f`: A function to apply to each element of `sgsʲs`.
+- `sgsʲs`: An iterator over the draft subdomain states.
 """
-    ρa⁺(gs)
+draft_sum(f, sgsʲs) = mapreduce_with_init(f, +, sgsʲs)
```

Comment:
I wonder whats the performance penalty for pretending we have more than 1 updraft and doing map reduce over 1 element?